src/app/core/entity-components/entity-subrecord/entity-subrecord/entity-subrecord.component.html

```diff
@@ -5,7 +5,7 @@
     matSort
     class="mat-elevation-z1 subrecord-table"
   >
-    <ng-container *ngFor="let col of _columns" [matColumnDef]="col.id">
+    <ng-container *ngFor="let col of getTableColumns()" [matColumnDef]="col.id">
```

Comment:
Lets put the filtered columns into a variable so the function is not executed to often.